common/classes/com/arjuna/common/internal/util/ClassloadingUtility.java

```diff
@@ -141,8 +141,17 @@ public static <T> T loadAndInstantiateClass(Class<T> iface, String className, St
             }
             commonLogger.logger.warn(message);
         }
+		catch (NoSuchMethodException e) {
+            String message = commonLogger.i18NLogger.warn_common_ClassloadingUtility_5(className, e);
```

Comment:
Where the existing log messages have been re-used, it may be that the wording is not appropriate for the new exception being checked. For example this new code uses https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/logging/arjunaI18NLogger.java#L879-L881 which is about "cannot access" which is appropriate for IllegalAccessException but I am not super sure it applies to NoSuchMethodException. WDYT @jmfinelli ?

Reply:
> Where the existing log messages have been re-used, it may be that the wording is not appropriate for the new exception being checked. For example this new code uses https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/logging/arjunaI18NLogger.java#L879-L881 which is about "cannot access" which is appropriate for IllegalAccessException but I am not super sure it applies to NoSuchMethodException. WDYT @jmfinelli ?

+1

Reply:
Thanks @tomjenkinson and @jmfinelli i will do the suggested changes.

Reply:
@tomjenkinson @jmfinelli what message i should use here for `NoSuchMethodException`?  As NoSuchMethodException is a class-loading issue should i use the same "cannot access" or i need to add a new message in `arjunaI18NLogger.java` WDYT?

Reply:
It seems there isn't any logging message that fits `NoSuchMethodException` in `arjunaI18NLogger`. Could you please create a new message? Thanks

Reply:
Unfortunately I think it might need a new message as the messages for catches do seem to be quite specific. 

In this case it would be something like "cannot create new instance of {0} as no zero-argument constructor exists"

Reply:
Thanks @tomjenkinson and @jmfinelli